src/connector/benches/nexmark_integration.rs

```diff
@@ -0,0 +1,142 @@
+
+fn make_batch() -> Vec<SourceMessage> {
+    let mut generator = nexmark::EventGenerator::default()
+        .with_type_filter(nexmark::event::EventType::Bid)
+        .map(|e| match e {
+            nexmark::event::Event::Bid(bid) => bid, // extract the bid event
+            _ => unreachable!(),
+        })
+        .enumerate();
+
+    let message_base = SourceMessage {
+        split_id: "default".into(),
+        key: None,
+        payload: None,     // to be filled
+        offset: "".into(), // to be filled
+        meta: SourceMeta::Empty,
+    };
+
+    generator
+        .by_ref()
+        .take(1024)
+        .map(|(i, e)| {
+            let payload = serde_json::to_vec(&e).unwrap();
+            SourceMessage {
+                payload: Some(payload),
+                offset: i.to_string(),
+                ..message_base.clone()
+            }
+        })
+        .collect_vec()
+}
+
+fn make_data_stream() -> BoxSourceStream {
+    futures::future::ready(Ok(BATCH.clone()))
+        .into_stream()
+        .boxed()
+}
+
+fn make_parser() -> impl ByteStreamSourceParser {
+    let columns = [
+        ("auction", DataType::Int64),
+        ("bidder", DataType::Int64),
+        ("price", DataType::Int64),
+        ("channel", DataType::Varchar),
+        ("url", DataType::Varchar),
+        ("date_time", DataType::Timestamp),
+        ("extra", DataType::Varchar),
+    ]
+    .into_iter()
+    .enumerate()
+    .map(|(i, (n, t))| SourceColumnDesc::simple(n, t, ColumnId::new(i as _)))
+    .collect_vec();
+
+    let props = SpecificParserConfig::DEFAULT_PLAIN_JSON;
+
+    JsonParser::new(props, columns, Default::default()).unwrap()
+}
+
+fn make_stream_iter() -> impl Iterator<Item = StreamChunkWithState> {
+    let mut stream: BoxSourceWithStateStream =
+        make_parser().into_stream(make_data_stream()).boxed();
+
+    std::iter::from_fn(move || {
+        stream
+            .try_next()
+            .now_or_never() // there's actually no yield point
+            .unwrap()
+            .unwrap()
+            .unwrap()
+            .into()
+    })
+}
+
+fn bench(c: &mut Criterion) {
+    let dispatch: tracing::dispatcher::Dispatch = tracing_subscriber::registry()
+        .with(tracing_subscriber::fmt::layer())
+        .with(
+            tracing_subscriber::filter::Targets::new()
+                .with_target("risingwave_connector", Level::INFO),
+        )
+        .into();
+
+    // Enable tracing globally.
+    //
+    // TODO: we should use `tracing::with_default` to set the dispatch in the scope,
+    // so that we can compare the performance with/without tracing side by side.
+    // However, why the global dispatch performs much worse than the scoped one?
+    dispatch.init();
+
+    c.bench_function("parse_nexmark", |b| {
+        b.iter_batched(
+            make_stream_iter,
+            |mut iter| iter.next().unwrap(),
+            BatchSize::SmallInput,
+        )
+    });
```

Comment:
Should it be like this?
```suggestion
    tracing::dispatcher::with_default(&dispatch, || {
        c.bench_function("parse_nexmark_with_tracing_scoped", |b| {
            b.iter_batched(
                make_stream_iter,
                |mut iter| iter.next().unwrap(),
                BatchSize::SmallInput,
            )
        })
    });
```

Reply:
If we only have single bench function of `parse_with_tracing_scoped`, then it's enabled correctly. 😄 If two are compiled together, then it seems to be disabled.

<img width="930" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/25862682/a343e6a4-2d25-4de7-aa06-57094492bfff">

Reply:
Resolved in #456cd32, but not sure why. 😄